### PR TITLE
Sentinel: App - Initial Submission

### PR DESCRIPTION
**Pull request by dfederschmidt**
#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [x] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [x] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [x] If this is a new connector or you are adding new actions
    - [x] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [x] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [x] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [x] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Initial release with incident-related actions and the ability to query the associated Log Analytics Workspace via `run query`

## Screenshots (if relevant)
Screenshot 2022-10-20 at 10 25 29

### CODE REVIEW

File: exclude_files.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.git*
```
Comment (dvora-crest) [inline exclude_files.txt:1]: Can we remove this file?

File: mssentinel.json [hunk 2]
```diff
@@ -0,0 +1,1072 @@
...
+    "description": "This app provides integration with Microsoft Sentinel",
+    "type": "information",
+    "product_vendor": "Microsoft",
+    "logo": "logo_mssentinel.svg",
+    "logo_dark": "logo_mssentinel_dark.svg",
+    "product_name": "Sentinel",
+    "python_version": "3",
+    "product_version_regex": ".*",
+    "publisher": "Splunk Community",
+    "license": "Copyright (c) Splunk Community, 2022",
```
Comment (dvora-crest) [inline mssentinel.json]: Shouldn’t the license value be "Copyright (c) 2022 Splunk Inc."?

File: readme.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
+<!-- File: readme.html
+  Copyright (c) 2022 Splunk Inc.
+
+  Licensed under Apache 2.0 (https://www.apache.org/licenses/LICENSE-2.0.txt)
```
Comment (dvora-crest) [inline readme.html]: Can we add the compelte license string?

File: mssentinel.json [hunk 4]
```diff
@@ -0,0 +1,1072 @@
...
+                }
+            },
+            "output": [],
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "get incident",
+            "identifier": "get_incident",
+            "description": "Gets a given incident",
+            "type": "generic",
```
Comment (dvora-crest) [inline mssentinel.json]: The action type contradicts with the read_only value. Could you please verify it for all the actions?

File: readme.html [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    <p>In order to retrieve the Workspace ID, navigate to your Sentinel Settings -> Workspace Settings</p>
+
+
+    <h1>Usage</h1>
+    <h2>How Sentinel handles identifiers</h2>
+    <p>Actions like <b>get incident</b> take an <i>incident name</i> input parameter. This can be captured from the Sentinel API or Web UI, but it's not to be confused with the Incident Number or the Title.
+      The Incident Name is the last component of the link to the incident that can be reviewed in Sentinel. For example, the Incident Name corresponding to
+    </p>
+    <p><code><pre>
+      https://portal.azure.com/#asset/Microsoft_Azure_Security_Insights/Incident/subscriptions/de582eca-4f28-4f8d-9ded-9b0cd2803739/resourceGroups/demomachine_group/providers/Microsoft.OperationalInsights/workspaces/customworkspace/providers/Microsoft.SecurityInsights/Incidents/80289647-8743-4a67-87db-9409e597b0db
```
Comment (dvora-crest) [inline readme.html]: It looks like actual ids are mistakenly added along with the URL, we can mask the last part of the ids with xxxxx

File: exclude_files.txt (hunk 1 above)
Comment (mmomin-crest) [reply exclude_files.txt:1]: The other temporary files [rest.http, .github/workflows and .gitignore] are also appended to the .tgz file if we remove this. I decided not to remove this file.

File: readme.html (hunk 5 above)
Comment (mmomin-crest) [reply readme.html]: Added dummy IDs.

File: readme.html (hunk 3 above)
Comment (mmomin-crest) [reply readme.html]: Yes, Added complete license string.

File: mssentinel.json (hunk 4 above)
Comment (mmomin-crest) [reply mssentinel.json]: Updated appropriate action types.

File: mssentinel.json (hunk 2 above)
Comment (mmomin-crest) [reply mssentinel.json]: Updated license value as per standard license string.

**Review by dvora-crest — APPROVED** [review]

**Pull request merged**